sensor/common/networkflow/manager/manager_impl.go

```diff
@@ -1081,16 +1337,17 @@ func (h *hostConnections) Process(networkInfo *sensor.NetworkConnectionInfo, now
 	}
 
 	{
-		prevSize := len(h.connections)
 		for c, t := range updatedConnections {
 			// timestamp = zero implies the connection is newly added. Add new connections, update existing ones to mark them closed
 			if t != timestamp.InfiniteFuture { // adjust timestamp if not zero.
 				t += tsOffset
 			}
-			status := h.connections[c]
-			if status == nil {
+			status, found := h.connections[c]
+			if !found || status == nil {
```

Comment:
Isn't this equivalent? I think if `found` is `false` then `status` is `nil`.

Reply:
You are right, I somehow thought that I need to make it clear that we do the same in either of those cases... but after giving it a second look, I think, I will correct it to the way it was before.